Add podman support

andylytical opened the pull request:
Changes:
- add parameter ignore_chain_prefixes
- add custom fact custom_firewallchains
- remove parameter purge_all

Tested on:
- jiracmdline.ncsa.illinois.edu

billglick reviewed (commented):
Generally seems good. A couple minor things:

- Add a note to `README` about `profile_firewall::purge_all` parameter being removed and no longer used
- Generate a new `REFERENCE.md` file

andylytical commented:
Updated `README.md` with a new section about ignoring chains, added a sample for Podman on RHEL8.
Also added some emoji for examples and notes to make them stand out a little better.

Generated a new `REFERENCE.md`

andylytical commented:
README updated with (re-tested) version of YAML that matches our standards!
